src/Promenade.Controllers/HomeController.cs

```diff
@@ -1,21 +1,362 @@
         }
 
+        [Route("")]
         public IActionResult Index()
         {
             return View();
         }
+        [HttpGet("[action]")]
+        [HttpGet("[action]/{address}")]
+        [HttpGet("[action]/{address}/{mobile}")]
+        public async Task<IActionResult> LocateZip(string address = null, int? mobile = null)
+        {
+            if (!string.IsNullOrWhiteSpace(address))
+            {
+                using (var client = new HttpClient())
+                {
+                    try
+                    {
+                        var apikey = _config["APIKey"];
+
+                        var response = await client.GetAsync($"https://maps.googleapis.com/maps/api/geocode/json?address={address}&key={apikey}");
+                        response.EnsureSuccessStatusCode();
+
+                        var stringResult = await response.Content.ReadAsStringAsync();
+                        dynamic jsonResult = JsonConvert.DeserializeObject(stringResult);
+
+                        if (jsonResult.results.Count > 0)
+                        {
+                            var result = jsonResult.results[0];
+                            double latitude = result.geometry.location.lat;
+                            double longitude = result.geometry.location.lng;
+
+                            Location viewModel = LibraryLookup(latitude, longitude);
+                            viewModel.FormattedAddress = result.formatted_address;
+
+                            return View("Locations",viewModel);
+                        }
+                        else
+                        {
+                            _logger.LogError($"No geocoding results for address \"{address}\"");
+                            TempData["AlertDanger"] = $"Unable to locate address <strong>\"{address}\"</strong>.";
+                        }
+                    }
+                    catch (HttpRequestException ex)
+                    {
+                        _logger.LogCritical(ex, $"Google API error: {ex.Message}");
+                        TempData["AlertDanger"] = "An error occured, please try again later.";
+                    }
+                    catch (Exception ex)
+                    {
+                        _logger.LogCritical(ex, ex.Message);
+                        TempData["AlertDanger"] = "An error occured, please try again later.";
+                    }
+                }
+            }
+            if (mobile == 1)
+            {
+                return View(new Location { ShowLocation = true });
+            }
+            return View("Locations");
+        }
+
+        [HttpGet("Locations")]
+        [HttpGet("Locations/[action]")]
+        public async Task<IActionResult> Find(double latitude = 0, double longitude = 0, string address = null, int? mobile = null)
+        {
+            Location[] locations = {};
+
+            if((latitude!=0 && longitude!=0) && address==null)
```

Comment:
1. Please use Edit -> Advanced -> Format Document (or `^E ^D`) on these files before checking in to ensure that they are formatted uniformly.
2. Strings should always be checked with `string.IsNullOrEmpty()` or `string.IsNullOrWhitespace()`, not `== null`